File: app/src/main/java/com/kirchhoff/example/githubclient/ui/LoadingDialog.java
package com.kirchhoff.example.githubclient.ui;

import android.app.DialogFragment;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;

import com.kirchhoff.example.githubclient.R;

/**
 * @author Kirchhoff-
 */

public class LoadingDialog extends DialogFragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        getDialog().getWindow().setBackgroundDrawable(new ColorDrawable(0));
        getDialog().getWindow().requestFeature(Window.FEATURE_NO_TITLE);
        getDialog().setCanceledOnTouchOutside(false);
        return inflater.inflate(R.layout.d_loading, container, false);
    }

}

File: app/src/main/java/com/kirchhoff/example/githubclient/ui/general/LoadingDialog.kt
package com.kirchhoff.example.githubclient.ui.general

import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog
import android.view.View

import com.kirchhoff.example.githubclient.R

/**
 * @author Kirchhoff-
 */

class LoadingDialog : DialogFragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        isCancelable = false
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(activity)
                .setView(View.inflate(activity, R.layout.d_loading, null))
                .create()
    }


}
